rwkv.cpp

```diff
@@ -7,24 +7,37 @@
 
 #include <string>
 #include <vector>
-#include <thread>
-#include <cassert>
+#include <cstring>
 #include <cinttypes>
 #include <cmath>
-#include <cstdio>
-#include <cstring>
 #include <fstream>
-#include <iostream>
 #include <unordered_map>
 #include <memory>
 
-#include <sys/stat.h> // fstat
+#define _FILE_OFFSET_BITS 64
+#define RWKV_MAYBE_BREAK
+
+#ifdef _MSC_BUILD
+#define stat _stat64
+#define fstat _fstat64
+#define ftell _ftelli64
+#define fseek _fseeki64
 
-#ifdef WIN32
-#define stat64 _stat64
-#define fstat64 _fstat64
+#ifndef NDEBUG
+#include <intrin.h>
+#define RWKV_MAYBE_BREAK __debugbreak()
+#endif
+#else
+#include <sys/stat.h>
+#if !defined(__APPLE__)
+#define ftell ftello
+#define fseek fseeko
+#endif
 #endif
 
+static_assert(sizeof(stat::st_size) >= 8, "file offsets should be 64-bit");
+static_assert(sizeof(decltype(ftell(NULL))) >= 8, "file offsets should be 64-bit");
```

Comment:
What this does exactly?

Reply:
this asserts that it supports files over 2GB

Reply:
I then propose adding a clarifying comment to the asserts :)

Reply:
I think I actually did when you made this comment, the more recent commits made the error messages a bit more descriptive

Reply:
Nvm, you've already changed the messages, I'm ok with them